consensus/src/dag/tests/dag_test.rs

```diff
@@ -82,14 +134,48 @@ fn test_dag_insertion_failure() {
     assert!(dag.add_node(node).is_err());
 }
 
+#[test]
+fn test_dag_recover_from_storage() {
+    let (signers, validator_verifier) = random_validator_verifier(4, None, false);
+    let epoch_state = Arc::new(EpochState {
+        epoch: 1,
+        verifier: validator_verifier.clone(),
+    });
+    let storage = Arc::new(MockStorage::new());
+    let mut dag = Dag::new(epoch_state.clone(), storage.clone());
+
+    let mut digests = vec![];
+
+    for round in 1..10 {
```

Comment:
do we treat that the rounds starts at 1 not 0?

Reply:
it's just simpler to manager when the dag is empty, I treat the lowest/highest round = 0, so round 1 can pass the rounds check otherwise I'd need to specialize it